plugins/inputs/http_response/http_response.go

```diff
@@ -148,6 +154,7 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 
 	fields["response_time"] = time.Since(start).Seconds()
 	fields["http_response_code"] = resp.StatusCode
+	fields["connection_failed"] = 0
```

Comment:
I think it is a good idea to always return a status whether the connection failed or not. If someone (i.e. me) wants to gather the latest value of the status and it is only returned when an error occured the value is always "error". 

Do we have a plan if we want to combine the fields and if so: how?

I can change the int fields to boolean if you want, however that would mean a breaking change for the response_match field.